### DISCUSSION TITLE
About `null`

### DISCUSSION THREAD

**Discussion by hchiba1** [pg-format/pg-formatter#39] [General]
What are the use cases for `null`?

**Comment by hchiba1** [comment]
What happens if the following is converted to PG-JSONL?
```
a :null p:null
```
Currently, it looks like:
```
{"type": "node", "id": "a", "labels": ["null"], "properties": {"p": [null]}}
```
`labels` are converted as expected. Not sure about `properties`.

**Comment by hchiba1** [comment]
One use case might be to display empty properties explicitly, so that the data schema is visible.

Considering that, it might be more natural to convert the example PG above, as follows:
```
{"type": "node", "id": "a", "labels": ["null"], "properties": {"p": []}}
```
but the code may become cumbersome for the operation of empty arrays instead of `null` values.

**Comment by hchiba1** [comment]
What happens if `null` is removed from the grammar?

**Reply by nichtich** [reply]
Whether `null` should be part of the data model or not is a choice of design. Would you prefer to remove support of `null` values? Most other graph data formats don't have them. In Cypher `key:null` is allowed but equivalent to omitting `key`.

**Reply by nichtich** [reply]
I'm ok with removing `null`. Both [PG data model](https://github.com/pg-format/specification?tab=readme-ov-file#property-graphs) and [PG-JSON](https://github.com/pg-format/specification?tab=readme-ov-file#pg-json) should then get fixed. There are open details of PG format grammar and of PG-JSONL but the common graph model and its serialization in PG-JSON seem to be quite stable.

**Reply by hchiba1** [reply]
Yes, I feel `null` is not necessarily easy to understand.
JavaScript beginners are often confused by `null` and `undefined`.
I would prefer eliminating such a source of confusion, if there is no specific benefit.

**Reply by nichtich** [reply]
Updated at https://github.com/pg-format/specification and in [pgraphs](https://www.npmjs.org/package/pgraphs) 0.5.0.
